### PR TITLE
changed constraints of delete button

### LINKED ISSUE #4852: "Profile Deletion" button is misaligned in landscape on a phone
**Describe the bug**
The "Profile Deletion" button in the "edit profile" screen on mobile landscape is overlapping with the other buttons available in the screen.

**To Reproduce**
1. Make sure there is more than one profile on the device.
1. Open Administrator Controls
2. Under "Profile Management" click "Edit profiles"
3. Select a **non-admin** profile
4. Rotate to landscape
5. Observe that the "Profile Deletion" button is overlapping the other buttons

**Expected behavior**

**Demonstration**
image

**Environment**
- Device/emulator being used: Nexus 5X
- Android or SDK version (e.g. Android 5 or SDK 21): SDK 27
- App version (you can get this through system app settings or via the admin controls menu in-app): an oppia_android_alpha_kenya build on tip-of-tree (71e769bec2b853d51996432e31f8bec232f16928), though I suspect this will also repro with oppia_dev.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by lonewolf2208**
## Explanation
Fixes #4852 
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

For landscape mode 
Before : 
image
After : 
image
image

### CODE REVIEW

**Comment by lonewolf2208** [comment]
Hey @BenHenning , Can you please review my changes .

**Comment by BenHenning** [comment]
Thanks @lonewolf2208, sorry about the delay in review here--I've been a bit slammed with a release that just took place.

Per the code, it seems like this is actually changing the placement of the button in landscape to be at the end of the button list rather than anchored to the bottom of the screen, as it is in portrait mode. Might it be possible to do that, instead, so that portrait and landscape are consistent?

**Comment by BenHenning** [comment]
Updating the branch & running CI so that we can get CI results for this PR.

**Comment by lonewolf2208** [comment]
> Thanks @lonewolf2208, sorry about the delay in review here--I've been a bit slammed with a release that just took place.
> 
> Per the code, it seems like this is actually changing the placement of the button in landscape to be at the end of the button list rather than anchored to the bottom of the screen, as it is in portrait mode. Might it be possible to do that, instead, so that portrait and landscape are consistent?

Good day @BenHenning , I think the button need to be constrained with the end buttons otherwise there would be problem with the small screens as it would be overlapping with it and further there is one more constraint from the bottom that needs to be removed as in case of tablet sized screen the button will be in middle if its constraint from bottom and end both .

**Comment by BenHenning** [comment]
> > Thanks @lonewolf2208, sorry about the delay in review here--I've been a bit slammed with a release that just took place.
> > Per the code, it seems like this is actually changing the placement of the button in landscape to be at the end of the button list rather than anchored to the bottom of the screen, as it is in portrait mode. Might it be possible to do that, instead, so that portrait and landscape are consistent?
> 
> Good day @BenHenning , I think the button need to be constrained with the end buttons otherwise there would be problem with the small screens as it would be overlapping with it and further there is one more constraint from the bottom that needs to be removed as in case of tablet sized screen the button will be in middle if its constraint from bottom and end both .

That's a good point @lonewolf2208, but I think we should rule out the technical feasibility, first. Could we maybe constrain the button both to the bottom of Reset PIN and to the bottom of the parent, then use a vertical constraint bias of 1 to push it toward the bottom of the screen? This will need to be checked on smaller screens (or with display zoom), but I think it would address the concern without changing the current UI.

**Comment by lonewolf2208** [comment]
Hey @BenHenning , I think adding vertical bias of 1 did the job . here are the snapshots from a tablet and a phone .

Tablet -
image
 Phone -  
image

can you please review the changes .

**Comment by BenHenning** [comment]
Ah sorry, I apparently missed this since Oppiabot didn't assign it. FYI @lonewolf2208 if you use "PTAL @<name>" then Oppiabot *should* auto-assign the PR.

**Comment by BenHenning** [comment]
I think the changes look solid @lonewolf2208. Could you please update your PR description to include the portrait screenshots, as well, for before & after?

Assigning this over to @rt4914 who will perform the codeowners review for it.

**Comment by lonewolf2208** [comment]
> I think the changes look solid @lonewolf2208. Could you please update your PR description to include the portrait screenshots, as well, for before & after?
> 
> Assigning this over to @rt4914 who will perform the codeowners review for it.

Done

**Comment by lonewolf2208** [comment]
Good day @rt4914 , Please examine the modifications and close this PR.

File: app/src/main/res/layout-land/profile_edit_fragment.xml
Comment (adhiamboperes) [inline app/src/main/res/layout-land/profile_edit_fragment.xml]: Is there a need to change the margin bottom here?
If we are changing it then how about updating the value of the `dimen/profile_edit_delete_margin_bottom`.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
@lonewolf2208 PTAL, I have left a comment.

Please update the branch as well from upstream/develop.

File: app/src/main/res/layout-land/profile_edit_fragment.xml
Comment (lonewolf2208) [reply app/src/main/res/layout-land/profile_edit_fragment.xml]: Sure will update the value of it

**Comment by rt4914** [comment]
@MohitGupta121 Can you please verify this PR? Also, I think the profile deletion should should not be just below "RESET PIN", instead there should be gap and it should be almost around bottom.

Nevermind, I went through Ben's comment, so will be reviewing it directly.

File: app/src/main/res/layout-land/profile_edit_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout-land/profile_edit_fragment.xml:275]: Add `app:layout_constraintEnd_toEndOf="parent"` too

**Review by rt4914 — APPROVED** [review]
@lonewolf2208 One minor change suggested.

File: app/src/main/res/layout-land/profile_edit_fragment.xml
Comment (rt4914) [reply app/src/main/res/layout-land/profile_edit_fragment.xml:275]:
```suggestion
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
```

**Review by adhiamboperes — APPROVED** [review]

**Comment by MohitGupta121** [comment]
> @MohitGupta121 Can you please verify this PR? 

@rt4914 Yes sure, I'll review this PR and verify the alignment of these 3 buttons.

**Comment by MohitGupta121** [comment]
@rt4914 Yes you are right there is Gap between "Reset Pin" and "Profile Deletion" as per the mocks and the latest code of develop branch.

Landscape [Mock](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/5b8a8202-6081-4855-8c49-94a8a5b4c559/specs/)

Tablet [Mock](https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/ab5bac96-af36-4ff2-bc5d-cc22f6255408/specs/)

Yes, I think we need to move "Profile Deletion" to the bottom according to mocks.

**Comment by adhiamboperes** [comment]
> @rt4914 Yes you are right there is Gap between "Reset Pin" and "Profile Deletion" as per the mocks and the latest code of develop branch.
> 
> Landscape [Mock](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/5b8a8202-6081-4855-8c49-94a8a5b4c559/specs/)
> 
> Tablet [Mock](https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/ab5bac96-af36-4ff2-bc5d-cc22f6255408/specs/)
> 
> Yes, I think we need to move "Profile Deletion" to the bottom according to mocks.

Hi @MohitGupta121, could you clarify this for me? Does it mean that there is something @lonewolf2208 still needs to do on this PR before we merge?

**Comment by MohitGupta121** [comment]
@adhiamboperes yes @lonewolf2208 need to keep "Profile Deletion" at the bottom of screen in Landscape mode. As per the the mocks I shared in this [comment](https://github.com/oppia/oppia-android/pull/4854#issuecomment-1464995453)

**Comment by adhiamboperes** [comment]
> @adhiamboperes yes @lonewolf2208 need to keep "Profile Deletion" at the bottom of screen in Landscape mode. As per the the mocks I shared in this [comment](https://github.com/oppia/oppia-android/pull/4854#issuecomment-1464995453)

Thanks @MohitGupta121 . @lonewolf2208 PTAL

**Comment by lonewolf2208** [comment]
> > @adhiamboperes yes @lonewolf2208 need to keep "Profile Deletion" at the bottom of screen in Landscape mode. As per the the mocks I shared in this [comment](https://github.com/oppia/oppia-android/pull/4854#issuecomment-1464995453)
> 
> Thanks @MohitGupta121 . @lonewolf2208 PTAL

Sure will make the changes

**Comment by rt4914** [comment]
UPDATE: 
As per https://github.com/oppia/oppia-android/pull/4854#issuecomment-1464872620 comment I did suggest adding gap between Profile deletion and Reset button earlier but at the end I wrote "Nevermind", as in ignore this comment because for landscape mode if we add gap between the profile deletion and reset button, it will end up either messing the UI like earlier (overlapping issue) or will introduce scrolling in the screen. So in both cases not uses gap is actually better.


Author just needs to make one final change and it should be merge ready:
https://github.com/oppia/oppia-android/pull/4854#discussion_r1133061568

**Pull request closed without merging**
